Src/ILGPU/IR/Construction/Structures.cs

```diff
@@ -80,6 +84,34 @@ partial class IRBuilder
                 else
                 {
                     instanceBuilder.Add(fieldValue);
+
+                    // If the field value we just added was the start of a fixed buffer,
+                    // copy the remaining elements, byte by byte.
+                    if (field.GetCustomAttribute<FixedBufferAttribute>() != null)
+                    {
+                        var fieldTypeInfo = typeInfo.GetFieldTypeInfo(i);
+                        var initialBytes = fieldValue.Type.Size;
+                        var numBytes = fieldTypeInfo.Size;
+                        var handle = GCHandle.Alloc(rawFieldValue, GCHandleType.Pinned);
+                        try
+                        {
+                            unsafe
+                            {
+                                byte* ptr = (byte*)handle.AddrOfPinnedObject();
+                                for (int j = initialBytes; j < numBytes; j++)
+                                {
```

Comment:
QUESTION: Do I need to worry about endian-ness?

Reply:
I guess this should work 🤓